[INFO -- 2018-12-08 05:47:09]
Starting flip_value script
version: 1.1
[INFO -- 2018-12-08 05:47:09]
Initialization strings:
[INFO -- 2018-12-08 05:47:09]
(1) file /tmp/knapsack/sa
[INFO -- 2018-12-08 05:47:09]
initSignal:0
[INFO -- 2018-12-08 05:47:09]
endSignal:9
[INFO -- 2018-12-08 05:47:09]
faultModel:1
[EXCEPTION -- 2018-12-08 05:47:15]
pythonException: empty range for randrange() (0,0, 0)
[EXCEPTION -- 2018-12-08 05:47:15]
pythonException: empty range for randrange() (0,0, 0)
[INFO -- 2018-12-08 05:47:16]
Backtrace BEGIN:
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff6ede801 in __GI_abort () at abort.c:79
#2  0x00007ffff6f27897 in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7ffff7054b9a "%s\n") at ../sysdeps/posix/libc_fatal.c:181
#3  0x00007ffff6f2e90a in malloc_printerr (str=str@entry=0x7ffff7052d88 "free(): invalid pointer") at malloc.c:5350
#4  0x00007ffff6f35e1c in _int_free (have_lock=0, p=0x555555771834, av=0x7ffff7289c40 <main_arena>) at malloc.c:4157
#5  __GI___libc_free (mem=0x555555771844) at malloc.c:3124
#6  0x0000555555556d66 in __gnu_cxx::new_allocator<int>::deallocate (this=0x555555759060 <val2>, __p=0x555555771844) at /usr/include/c++/7/ext/new_allocator.h:125
#7  0x0000555555556912 in std::allocator_traits<std::allocator<int> >::deallocate (__a=..., __p=0x555555771844, __n=263) at /usr/include/c++/7/bits/alloc_traits.h:462
#8  0x0000555555556176 in std::_Vector_base<int, std::allocator<int> >::_M_deallocate (this=0x555555759060 <val2>, __p=0x555555771844, __n=263) at /usr/include/c++/7/bits/stl_vector.h:180
#9  0x0000555555555eb1 in std::_Vector_base<int, std::allocator<int> >::~_Vector_base (this=0x555555759060 <val2>, __in_chrg=<optimized out>) at /usr/include/c++/7/bits/stl_vector.h:162
#10 0x0000555555555949 in std::vector<int, std::allocator<int> >::~vector (this=0x555555759060 <val2>, __in_chrg=<optimized out>) at /usr/include/c++/7/bits/stl_vector.h:435
#11 0x00007ffff6ee1041 in __run_exit_handlers (status=0, listp=0x7ffff7289718 <__exit_funcs>, run_list_atexit=run_list_atexit@entry=true, run_dtors=run_dtors@entry=true) at exit.c:108
#12 0x00007ffff6ee113a in __GI_exit (status=<optimized out>) at exit.c:139
#13 0x000055555555504b in exit_gracefully () at knapsacksa.cpp:40
#14 0x0000555555555200 in cost (solution=std::vector of length 200, capacity 200 = {...}) at knapsacksa.cpp:66
#15 0x000055555555534c in simulated_annealing () at knapsacksa.cpp:92
#16 0x0000555555555658 in main () at knapsacksa.cpp:146

46	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.

Backtrace END
Fliping value of field: __val
Fliping array at pos: 10
Memory content before bitflip:0xff0xff0xff0xff0x000x000x000x00
Fault Model: Double bit-flip
base address to flip value: 0x7fffffffcf80
address max offset: 8
Memory content after  bitflip:0xff0xff0xff0xff0x000x0a0x000x00
frame name: __GI_raise
symbol name: set
symbol filename: ../sysdeps/unix/sysv/linux/raise.c
symbol line: 39
value: {__val = {0, 140737353962768, 140737346239191, 8453489135, 140737488342912, 140737488342928, 140737353963624, 0, 1, 140737353962768, 10999411245055, 4294967297, 140737335955016, 140737353966560, 140737353963624, 0}}
value address: 0x7fffffffcf30
Type: The type is a structure.
Type sizeof: 128
Field name: __val
Field Type: The type is an array.
Field Type sizeof: 128

[INFO -- 2018-12-08 05:47:16]
Fault Injection Successful after 7.020445108413696s
[INFO -- 2018-12-08 05:47:16]
event type: exit
[EXCEPTION -- 2018-12-08 05:47:16]
exit code: no exit code available
